test/sparse/sparse.jl

```diff
@@ -1659,3 +1659,7 @@ let X = sparse([1 -1; -1 1])
         @test Y / 1 == Y
     end
 end
+
+# Check that `broadcast` methods specialized for unary operations over
+# `SparseMatrixCSC`s determine a reasonable return type. (Issue #18974.)
+@test eltype(sin.(spdiagm(Int64(1):Int64(4)))) == Float64
```

Comment:
Derp on my part explains that: The first version was `isa(eltype(...), Float64)`, with the failure due to `isa`. Best!